src/persist/src/postgres.rs

```diff
@@ -404,6 +412,33 @@ mod tests {
             }
         };
 
-        consensus_impl_test(|| PostgresConsensus::open(config.clone())).await
+        consensus_impl_test(|| PostgresConsensus::open(config.clone())).await?;
+
+        // and now verify the implementation-specific `drop_and_recreate` works as intended
+        let consensus = PostgresConsensus::open(config.clone()).await?;
+        let deadline = Instant::now() + Duration::from_secs(5);
+        let key = Uuid::new_v4().to_string();
+        let state = VersionedData {
+            seqno: SeqNo(5),
+            data: Bytes::from("abc"),
+        };
+
+        assert_eq!(
+            consensus
+                .compare_and_set(deadline, &key, None, state.clone())
+                .await,
+            Ok(Ok(()))
+        );
+
+        assert_eq!(
+            consensus.head(deadline, &key).await,
+            Ok(Some(state.clone()))
+        );
+
+        consensus.drop_and_recreate().await?;
+
+        assert_eq!(consensus.head(deadline, &key).await, Ok(None));
+
+        Ok(())
```

Comment:
I wanted this whole addendum to this test to be its own unit test, but then we run into issues with the tests executing in parallel and deleting each other's state... seemed simplest to just bolt it on